tmt/base.py

```diff
@@ -980,6 +989,37 @@ def _save_tree(self, tree):
         except tmt.utils.MetadataError:
             self._use_default_plan()
 
+    def _copy_tree_root(self):
+        """ Copy tree root to the workdir, under 'tree' directory """
+        tree_root = self.tree.root or '.'
+        workdir_tree = self.workdir_tree
+        # Use git worktree to initialize the tree
+        if os.path.exists(os.path.join(tree_root, '.git')):
+            try:
+                self._git_worktree(tree_root)
+                # Make sure tree is not is not created again in dest
+                workdir_tree = f'{workdir_tree}/'
+            except tmt.utils.RunError as error:
+                self.debug(f'Not using git worktree due to: {error}')
+                pass
+
+        # Sync additional data, except .git
+        self.run(
+            f'rsync -arv --exclude .git {tree_root}/ {workdir_tree}',
+            cwd=self.workdir)
+
+    def _git_worktree(self, tree_root):
+        """ Create git worktree under 'tree' directory """
```

Comment:
If we are going to use the `worktree` feature, it seems we will need to take care of the cleanup as well. I did some experimenting and after a few attempts I resulted with this:

```
> git branch
* master
+ var-tmp-tmt-run-330
+ var-tmp-tmt-run-331
+ var-tmp-tmt-run-332
```

Plus, if the run directory is removed, the worktree stays there, so the next time you get:

```
Not using git worktree due to: ("Failed to run command 'git -C /tmp/mini worktree add -b var-tmp-tmt-run-332 /var/tmp/tmt/run-332/tree'.", 'git -C /tmp/mini worktree add -b var-tmp-tmt-run-332 /var/tmp/tmt/run-332/tree', 255, "Preparing worktree (new branch 'var-tmp-tmt-run-332')\n", "fatal: A branch named 'var-tmp-tmt-run-332' already exists.\n")
```

Would it make sense to start with just copying files as the first step to fix the problem?

Reply:
we can just cleanup in finish, not a big problem

Reply:
of course, if somebody sees a problem with that, we can just do rsync